flocker/common/test/test_algebraic.py

```diff
@@ -0,0 +1,117 @@
+# -*- test-case-name: flocker.common.test.test_algebraic -*-
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Tests for ``flocker.common.algebraic``
+"""
+
+from pyrsistent import (
+    PClass, field,
+    InvariantException,
+)
+from hypothesis import given, strategies as st, assume
+
+from twisted.trial.unittest import SynchronousTestCase
+
+from twisted.python.constants import Names, NamedConstant
+
+
+from ..algebraic import StateInvariant
+
+
+class States(Names):
+    ALLOWED = NamedConstant()
+    WITH_ATTRIBUTE = NamedConstant()
+    WITH_TWO_ATTRIBUTES = NamedConstant()
+    DISALLOWED = NamedConstant()
+
+
+class AlgebraicType(PClass):
+    state = field(mandatory=True)
+    one = field(bool)
+    two = field(bool)
+
+    __invariant__ = StateInvariant(
+        allowed_states={
+            States.ALLOWED,
+            States.WITH_ATTRIBUTE,
+            States.WITH_TWO_ATTRIBUTES,
+        },
+        expected_attributes={
+            'one': {States.WITH_ATTRIBUTE, States.WITH_TWO_ATTRIBUTES},
+            'two': {States.WITH_TWO_ATTRIBUTES},
+        },
+    )
+
+ARGS_STRATEGEY = st.fixed_dictionaries({
```

Comment:
Spelling: "strategy", not "strategey".

Although if you make the changes I suggest below (essentially making a strategy to generate valid arguments), then I think this probably won't be necessary any more.